### PR TITLE
Add ohai_hint resource from ohai cookbook

### PR DESCRIPTION
**Pull request by tas50**
This makes it easier to drop off a hint and reload ohai w/o bringing in cookbook dependencies.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/chef/resource/ohai_hint.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      introduced "14.0"
+
+      property :hint_name,
+               String,
+               description: "The name of hints file if different from the resource name",
+               name_property: true
+
+      property :content,
+               Hash,
+               description: "A Hash of values to include in the hint file"
```
Comment (thommay) [inline lib/chef/resource/ohai_hint.rb]: > Data to include in the hint file

We already know it's a hash, and our docs should show the type and the description.

File: lib/chef/resource/ohai_hint.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+# limitations under the License.
+#
+
+class Chef
+  class Resource
+    class OhaiHint < Chef::Resource
+      resource_name :ohai_hint
+      provides :ohai_hint
+
+      description "Use the ohai_hint resource to pass hint data to Ohai to aid in configuration detection."
```
Comment (thommay) [inline lib/chef/resource/ohai_hint.rb]: Don't think we should repeat the name of the resource in the description. We could probably use some wordsmithing and advice from @brewn here. (And on the rest of these).

Comment (tas50) [reply lib/chef/resource/ohai_hint.rb]: This goes off the format on the docs site. Here's an example:

Use the apt_repository resource to specify additional APT repositories. Adding a new repository will update APT package cache immediately.

**Comment by tas50** [comment]
@thommay Does this work with the wording fixes?

**Pull request merged**